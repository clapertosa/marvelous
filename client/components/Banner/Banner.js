import Carousel from "../Carousel/Carousel";
import styles from "./Banner.scss";

const Banner = () => {
  const banner = [
    "/static/backgrounds/banner0.jpg",
    "/static/backgrounds/banner1.jpg",
    "/static/backgrounds/banner2.jpg",
    "/static/backgrounds/banner3.jpg"
  ];
  return (
    <div className={styles.container}>
      <Carousel
        autoplay={{ delay: 3000 }}
        slidesPerView={1}
        allowTouchMove={false}
        disableOnInteraction={false}
        loop
        speed={300}
      >
        {banner.map((image, index) => (
          <img key={index} src={image} />
        ))}
      </Carousel>
    </div>
  );
};

export default Banner;
