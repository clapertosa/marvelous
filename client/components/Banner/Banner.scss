@import "../../static/styles/_variables";

.container {
  position: relative;
  margin-top: 20px;
  margin-right: auto;
  margin-left: auto;
  width: 95%;
  overflow: hidden;
  z-index: 70;
}

@media (min-width: $mediaQueryMinWidth) {
}
