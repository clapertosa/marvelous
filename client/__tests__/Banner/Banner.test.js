import { shallow } from "enzyme";
import { toJson } from "enzyme-to-json";
import Banner from "../../components/Banner/Banner";

describe("Banner", () => {
  it("should correctly display backgrounds", () => {
    const wrapper = shallow(<Banner />);
    expect(wrapper.find("img")).toHaveLength(4);
    const srcs = wrapper.find("img").map(node => node.props().src);
    expect(srcs).toEqual([
      "/static/backgrounds/banner0.jpg",
      "/static/backgrounds/banner1.jpg",
      "/static/backgrounds/banner2.jpg",
      "/static/backgrounds/banner3.jpg"
    ]);
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<Banner />);
    expect(wrapper.toJson).toMatchSnapshot();
  });
});
