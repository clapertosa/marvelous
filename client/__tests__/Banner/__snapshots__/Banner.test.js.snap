// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should match the snapshot 1`] = `undefined`;
